src/units/unit.cpp

```diff
@@ -1482,8 +1482,26 @@ static bool matches_ability_filter(const config & cfg, const std::string& tag_na
 	if(!string_matches_if_present(filter, cfg, "active_on", "both"))
 		return false;
 
-	if(!int_matches_if_present(filter, cfg, "value"))
-		return false;
+	if(!filter["value"].empty()){
+		bool has_other_key = (cfg["add"].empty() && cfg["sub"].empty() && cfg["multiply"].empty() && cfg["divide"].empty()) ? false : true;
```

Comment:
```suggestion
		bool has_other_key = !(cfg["add"].empty() && cfg["sub"].empty() && cfg["multiply"].empty() && cfg["divide"].empty());
```
IMO that is easier to read. Perhaps even clearer might be: `!cfg["add"].empty() || !cfg["sub"].empty() ||` ...

Reply:
done